sdk/dotnet/Pulumi.Automation/ProjectRuntime.cs

```diff
@@ -15,5 +18,37 @@ public ProjectRuntime(ProjectRuntimeName name)
         {
             this.Name = name;
         }
+
+        internal static IEqualityComparer<ProjectRuntime> Comparer { get; } = new ProjectRuntimeComparer();
+
+        private sealed class ProjectRuntimeComparer : IEqualityComparer<ProjectRuntime>
+        {
+            bool IEqualityComparer<ProjectRuntime>.Equals(ProjectRuntime? x, ProjectRuntime? y)
+            {
+                if (x == null)
+                {
+                    return y == null;
+                }
+
+                if (y == null)
+                {
+                    return x == null;
+                }
+
+                return x.Name == y.Name && ProjectRuntimeOptions.Comparer.Equals(x.Options, y.Options);
+            }
+
+            int IEqualityComparer<ProjectRuntime>.GetHashCode(ProjectRuntime obj)
+            {
+                var hash = new HashCode();
+                hash.Add(obj.Name);                
+                if (obj.Options != null)
+                {
+                    hash.Add(ProjectRuntimeOptions.Comparer.GetHashCode(obj.Options));
+                }
+                return hash.ToHashCode();
```

Comment:
Can consider using [HashCode.Combine](https://docs.microsoft.com/en-us/dotnet/api/system.hashcode.combine?view=net-5.0)

In this case we default to 0 if Options are null, which HashCode.Combine also does

```suggestion
                return HashCode.Combine(
                    obj.Name,
                    obj.Options != null ? ProjectRuntimeOptions.Comparer.GetHashCode(obj.Options) : 0
                );
```